easybuild/framework/easyblock.py

```diff
@@ -3363,6 +3363,16 @@ def post_processing_step(self):
 
     def _dispatch_sanity_check_step(self, *args, **kwargs):
         """Decide whether to run the dry-run or the real version of the sanity-check step"""
+        if 'extension' in kwargs:
+            extension = kwargs.pop('extension')
+            self.log.deprecated(
+                "Passing `extension` to `sanity_check_step` is no longer necessary and will be ignored "
+                f"(Easyblock: {self.__class__.__name__}).",
+                '6.0',
+            )
+            if extension != self.is_extension:
+                raise EasyBuildError('Unexpected value for `extension` argument. '
```

Comment:
This isn't exactly ignoring it, as is stated above...

What's the point of raising an *error* for an incorrect value being passed that's going to be ignored?!